yeonjae/programmers-1/달리기경주.js

```diff
@@ -0,0 +1,39 @@
+// 객체를 이용한 풀이
+function solution(players, callings) {
+  const playerTemp = {};
+
+  players.forEach((player, idx) => (playerTemp[player] = idx));
+
+  for (let i = 0; i < callings.length; i++) {
+    const curPlayer = callings[i];
+    const curPlayerIdx = playerTemp[curPlayer];
+    const backPlayer = players[curPlayerIdx - 1];
+
+    players[curPlayerIdx - 1] = curPlayer;
+    players[curPlayerIdx] = backPlayer;
+
+    playerTemp[curPlayer] = curPlayerIdx - 1;
+    playerTemp[backPlayer] = curPlayerIdx;
+  }
+
+  return players;
+}
```

Comment:
```suggestion
function solution(players, callings) {
  const copyPlayers =[...players];
  const playerObj = {};

  copyPlayers.forEach((player, idx) => (playerObj[player] = idx));

  for (let i = 0; i < callings.length; i++) {
    const curPlayer = callings[i];
    const curPlayerIdx = playerTemp[curPlayer];
    const backPlayer = players[curPlayerIdx - 1];

    copyPlayers[curPlayerIdx - 1] = curPlayer;
    copyPlayers[curPlayerIdx] = backPlayer;

    playerObj[curPlayer] = curPlayerIdx - 1;
    playerObj[backPlayer] = curPlayerIdx;
  }

  return copyPlayers;
}
```
- 풀이잘봤습니다!!
- 객체를 활용한 풀이 잘봤습니다 약간 아쉬운점이라면 players를 ...연산자로 복사해서 이를 활용했으면 어떨까 싶네요!
- 또한, playerTemp보다는 어떤 타입의 데이터인지 명시해주는 playerObj가 더 나아보입니다~